extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -584,4 +815,79 @@ public TopicPartition apply(Integer n)
         )
     );
   }
+
+  private Set<Integer> assignPartitionsAndSeekToStart(KafkaConsumer consumer, String topic)
+  {
+    // Initialize consumer assignment.
+    final Set<Integer> assignment = Sets.newHashSet();
+    for (Map.Entry<Integer, Long> entry : nextOffsets.entrySet()) {
+      final long endOffset = endOffsets.get(entry.getKey());
+      if (entry.getValue() < endOffset) {
+        assignment.add(entry.getKey());
+      } else if (entry.getValue() == endOffset) {
+        log.info("Finished reading partition[%d].", entry.getKey());
+      } else {
+        throw new ISE(
+            "WTF?! Cannot start from offset[%,d] > endOffset[%,d]",
+            entry.getValue(),
+            endOffset
+        );
+      }
+    }
+
+    assignPartitions(consumer, topic, assignment);
+
+    // Seek to starting offsets.
+    for (final int partition : assignment) {
+      final long offset = nextOffsets.get(partition);
+      log.info("Seeking partition[%d] to offset[%,d].", partition, offset);
+      consumer.seek(new TopicPartition(topic, partition), offset);
+    }
+
+    return assignment;
+  }
+
+  /**
+   * Checks if the pauseRequested flag was set and if so blocks:
+   * a) if pauseMillis == 0, until pauseRequested is cleared and shouldResume is signalled
```

Comment:
I'm just reacting to seeing "blocks until pauseRequested is cleared and shouldResume is signalled". We can't really know if shouldResume was signalled or not. We can know if pauseRequested was cleared. If that's all we need to know then I think the current code is OK (but the comment is confusing)